sotopia/experimental/agents/moderator.py

```diff
@@ -68,43 +68,143 @@ def __init__(
         self.current_agent_index: int = 0
         self.scenario: str = scenario
         self.agents: list[str] = list(agent_mapping.values())
+        self.agent_models: dict[str, str] = {}
+        self.agents_awake: dict[str, bool] = {name: False for name in self.agents}
+        self.all_agents_awake: asyncio.Event = asyncio.Event()
+        self.message_history: list[list[tuple[str, str, str]]] = [
+            [("Environment", "Environment", self.scenario)]
+        ]
 
     async def send(self, action: Observations) -> None:
         for output_channel, output_channel_type in self.output_channel_types.items():
-            await self.r.publish(
-                output_channel,
-                Message[output_channel_type](
-                    data=action.observations_map[output_channel]
-                ).model_dump_json(),  # type:ignore[valid-type]
-            )
+            if output_channel in action.observations_map:
+                await self.r.publish(
+                    output_channel,
+                    Message[output_channel_type](  # type:ignore[valid-type]
+                        data=action.observations_map[output_channel]
+                    ).model_dump_json(),
+                )
 
     async def __aenter__(self) -> Self:
         print(self.scenario)
-        await self.send(
-            Observations(
+        asyncio.create_task(self.booting())
+        self.task_scheduler = asyncio.create_task(self._task_scheduler())
+        return await super().__aenter__()
+
+    async def _task_scheduler(self) -> None:
+        await self.all_agents_awake.wait()
+        return await super()._task_scheduler()
+
+    async def booting(self) -> None:
+        """
+        1. send checking message to agents for every 5 seconds, until all agents are awake
```

Comment:
can we not do like 0.1s? I think agents initiate super fast, 5s is a bit long for scaled simulations